File: composeApp/src/desktopMain/kotlin/fr/thomasbernard03/androidtools/presentation/fileexplorer/FileExplorerEvent.kt
package fr.thomasbernard03.androidtools.presentation.fileexplorer

import fr.thomasbernard03.androidtools.domain.models.File
import fr.thomasbernard03.androidtools.domain.models.Folder
import fr.thomasbernard03.androidtools.presentation.commons.Event

sealed class FileExplorerEvent : Event {
    data object OnAppearing : FileExplorerEvent()

    data class OnGetFiles(val folder : Folder) : FileExplorerEvent()

    data object OnGoBack : FileExplorerEvent()
    data object OnRefresh : FileExplorerEvent()

    data class OnAddFile(val path : String) : FileExplorerEvent()
    data class OnDelete(val path : String) : FileExplorerEvent()

    data class OnFileSelected(val file : File?) : FileExplorerEvent()
}

File: lib/features/fileexplorer/presentation/file_explorer_event.dart
part of 'file_explorer_bloc.dart';

sealed class FileExplorerEvent {}

class OnAppearing extends FileExplorerEvent {}

class OnGoToFolder extends FileExplorerEvent {
  final FileEntry folder;

  OnGoToFolder({required this.folder});
}

class OnGoBack extends FileExplorerEvent {}
